.github/workflows/release.yaml

```diff
@@ -23,31 +23,60 @@ on:
   push:
     branches:
       - main
+
+defaults:
+  run:
+    shell: nix develop --impure --command bash {0}
+
 jobs:
   release:
     runs-on: ubuntu-latest
     permissions:
       contents: write
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4 # v3.5.3
+      - uses: cachix/install-nix-action@v26
         with:
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      # Cache the nix store.
+      - uses: DeterminateSystems/magic-nix-cache-action@v4
+
+      # Try to checkout a branch with the same name as provided in the dispatch
+      - uses: actions/checkout@v4 # v3.5.3
+        id: checkout
+        with:
+          ref: ${{ github.event.client_payload.branch }}
           fetch-depth: 0
+          token: ${{ secrets.ACTIONS_BOT_TOKEN }}
+        continue-on-error: true
 
-      - name: Set up Helm
-        uses: azure/setup-helm@5119fcb9089d432beecbf79bb2c7915207344b78 # v3.5
+      # If we couldn't checkout the provided branch, we will checkout the default branch
+      # instead. We assume that the branch does not yet exist, but we will create it,
+      # when we commit the updated dependencies.
+      - uses: actions/checkout@v4 # v3.5.3
+        if: ${{ steps.checkout.outcome != 'success' }}
         with:
-          version: v3.12.2
+          token: ${{ secrets.ACTIONS_BOT_TOKEN }}
+          fetch-depth: 0
 
       - name: Add Repositories for each chart
         run: |
           helm repo add redpanda https://charts.redpanda.com
 
-      - name: Configure Git
+      - name: Update testdata
         run: |
-          git config user.name "$GITHUB_ACTOR"
-          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          go test ./charts/redpanda -update -short
 
+      # Commit changes must be run before chart-releaser execution, because auto commit would include helm chart tar.gz artifact
+      - name: Commit changes for testdata golden files
+        uses: EndBug/add-and-commit@6e9dd5ddd54bb6b0c1d7b21824951061b9d0a2c0
```

Comment:
Just my two cents, it might be better to just write a shell script that does this commit for us? It's one less dep for the project and we can run shell scripts locally to test them out. No need to change if this works though.

Reply:
I just borrowed this from console-enterpise repo.

I will change that to
```
          git config --global user.name 'vbotbuildovich'
          git config --global user.email 'vbotbuildovich@users.noreply.github.com'
          git config --global --add --bool push.autoSetupRemote true
          git add --all
          git commit -m "[Auto Commit] Update console-oss dependency"
          git push
```